### PR TITLE
Chore: add option to use 4th az

### PR DESCRIPTION
**Pull request by rdong333**

### CODE REVIEW

File: src/main/resources/cloudformation/vpc.yaml [hunk 1]
```diff
@@ -71,6 +79,13 @@ Parameters:
     MaxLength: '20'
     MinLength: '9'
     Type: String
+  subnetCidrBlockForAz4:
```
Comment (mayitbeegh) [inline src/main/resources/cloudformation/vpc.yaml:82]: Should be conditional

File: src/main/resources/cloudformation/vpc.yaml [hunk 2]
```diff
@@ -71,6 +79,13 @@ Parameters:
...
+    AllowedPattern: '[0-9./]*'
+    Default: 172.20.12.0/24
+    Description: Cidr block for subnet in AZ '3'
```
Comment (mayitbeegh) [inline src/main/resources/cloudformation/vpc.yaml:85]: 4 not 3

File: src/main/resources/cloudformation/vpc.yaml [hunk 3]
```diff
@@ -21,6 +22,9 @@ Outputs:
   vpcSubnetIdForAz3:
     Description: The VPC subnet in AZ '3'
     Value: !Ref 'CerberusSubnetForAz3'
+  vpcSubnetIdForAz4:
```
Comment (mayitbeegh) [inline src/main/resources/cloudformation/vpc.yaml:25]: Should be conditional

File: src/main/resources/cloudformation/vpc.yaml (hunk 1 above)
Comment (rdong333) [reply src/main/resources/cloudformation/vpc.yaml:82]: what would change though? this parameter won't be used if the az4 parameter isn't provided

File: src/main/resources/cloudformation/vpc.yaml [hunk 4]
```diff
@@ -43,6 +47,10 @@ Parameters:
     AllowedPattern: '[a-z]{2}-[a-z]+-\d\w'
     Description: An availability zone identifier for zone '3'
     Type: String
+  az4:
+    Default: ""
```
Comment (mayitbeegh) [inline src/main/resources/cloudformation/vpc.yaml:51]: Seems like it's missing allowed pattern

File: src/main/resources/cloudformation/vpc.yaml (hunk 3 above)
Comment (rdong333) [reply src/main/resources/cloudformation/vpc.yaml:25]: The condition is used in the "Value" field

File: src/main/resources/cloudformation/vpc.yaml (hunk 4 above)
Comment (rdong333) [reply src/main/resources/cloudformation/vpc.yaml:51]: So this was something I was having a problem with. If the allowed pattern is included, this parameter cannot be an empty string. I wasn't sure what would be a good way to include the pattern while also allowing this to be an empty string.

File: src/main/resources/cloudformation/vpc.yaml (hunk 1 above)
Comment (mayitbeegh) [reply src/main/resources/cloudformation/vpc.yaml:82]: You're right

File: src/main/resources/cloudformation/vpc.yaml (hunk 3 above)
Comment (mayitbeegh) [reply src/main/resources/cloudformation/vpc.yaml:25]: Does it output an empty string?

File: src/main/resources/cloudformation/vpc.yaml (hunk 4 above)
Comment (mayitbeegh) [reply src/main/resources/cloudformation/vpc.yaml:51]: That's fine then

File: src/main/resources/cloudformation/vpc.yaml (hunk 3 above)
Comment (rdong333) [reply src/main/resources/cloudformation/vpc.yaml:25]: Yes, if a value for the az4 parameter is not provided it outputs an empty string.

Comment (rdong333) [reply src/main/resources/cloudformation/vpc.yaml:25]: Yes, if a value for the az4 parameter is not provided it outputs an empty string.

**Review by mayitbeegh — APPROVED** [review]

**Pull request merged**